library/core/src/num/niche_types.rs

```diff
@@ -49,12 +53,14 @@ macro_rules! define_valid_range_type {
             /// Immediate language UB if `val == 0`, as it violates the validity
             /// invariant of this type.
             #[inline]
+            #[cfg_attr(flux, flux::spec(fn (val: $int{ $low <= cast(val) && cast(val) <= $high }) -> Self[{val:cast(val)}]))] // NOTE: `val == 0` comments are stale(?)
```

Comment:
I'd agree that `0` is not relevant here (anymore?) - would you mind creating an upstream PR that proposes to fix this?

Reply:
Sure thing, @nilehmann will you be able to do this?

Reply:
will do

Reply:
https://github.com/rust-lang/rust/pull/144797